testsuite/single_node_performance.py

```diff
@@ -239,61 +279,62 @@ def print_table(
 warnings = []
 
 with tempfile.TemporaryDirectory() as tmpdirname:
-    create_db_command = f"cargo run {BUILD_FLAG} -- --block-size {BLOCK_SIZE} --concurrency-level {CONCURRENCY_LEVEL} --split-ledger-db --use-sharded-state-merkle-db --skip-index-and-usage create-db --data-dir {tmpdirname}/db --num-accounts {NUM_ACCOUNTS}"
+    create_db_command = f"cargo run {BUILD_FLAG} -- --block-size {MAX_BLOCK_SIZE} --execution-threads {NUMBER_OF_EXECUTION_THREADS} {DB_CONFIG_FLAGS} create-db --data-dir {tmpdirname}/db --num-accounts {NUM_ACCOUNTS}"
     output = execute_command(create_db_command)
 
     results = []
 
     for (
-        test_index,
-        (
-            (transaction_type_name, use_native_executor, module_working_set_size),
-            (
-                expected_tps,
-                check_active,
-            ),
-        ),
-    ) in enumerate(EXPECTED_TPS.items()):
-        print(f"Testing {transaction_type_name}")
-        if transaction_type_name == "mix_publish_transfer":
-            transaction_type = "publish-package coin-transfer"
-            transaction_weights = "1 500"
+        test_index, 
+        test,
+    ) in enumerate(TESTS):
+        if SELECTED_FLOW not in test.included_in:
+            continue
+
+        print(f"Testing {test.key}")
+        if test.key.transaction_type_override == "":
+            workload_args_str = ""
         else:
-            transaction_type = transaction_type_name
-            transaction_weights = "1"
+            transaction_type_list = test.key.transaction_type_override or test.key.transaction_type
+            transaction_weights_list = test.key.transaction_weights_override or "1"
+            workload_args_str = f"--transaction-type {transaction_type_list} --transaction-weights {transaction_weights_list}"
 
-        cur_block_size = int(min([expected_tps, BLOCK_SIZE]))
+        cur_block_size = int(min([test.expected_tps, MAX_BLOCK_SIZE]))
 
-        executor_type = "native" if use_native_executor else "VM"
+        sharding_traffic_flags = test.key.sharding_traffic_flags or ""
 
-        use_native_executor_str = "--use-native-executor" if use_native_executor else ""
-        common_command_suffix = f"{use_native_executor_str} --generate-then-execute --transactions-per-sender 1 --block-size {cur_block_size} --split-ledger-db --use-sharded-state-merkle-db --skip-index-and-usage run-executor --transaction-type {transaction_type} --transaction-weights {transaction_weights} --module-working-set-size {module_working_set_size} --main-signer-accounts {MAIN_SIGNER_ACCOUNTS} --additional-dst-pool-accounts {ADDITIONAL_DST_POOL_ACCOUNTS} --data-dir {tmpdirname}/db  --checkpoint-dir {tmpdirname}/cp"
+        if test.key.executor_type == "VM":
+            executor_type_str = "--transactions-per-sender 1" 
```

Comment:
@msmouse wanted me to add [this](https://github.com/aptos-labs/aptos-core/blob/0056a7b570d3e31cb0218059092cc50434b11a29/execution/executor-benchmark/src/main.rs#L192C1-L194C1) to mark `--transactions-per-sender` and `--hotspot-probability` conflicting with each other...

Reply:
```
error: the argument '--transactions-per-sender <TRANSACTIONS_PER_SENDER>' cannot be used with '--hotspot-probability <HOTSPOT_PROBABILITY>'
```

Reply:
I don't know, maybe just drop the constraint and swallow it?

Reply:
thanks for raising!

for this PR that is ok - as test.key.executor_type == "sharded" is where we don't add --transactions-per-sender.

but will follow-up:
- is executor type a right place to switch between sharded and not sharded, i.e. can we use sharding with native executor?
- we might want to use --hotspot-probability with non-sharded workload - to have identical workload for testing, which this doesn't allow